<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="configuration_basename" default="simulation.lua"/>
  <arg name="open_rviz" default="true"/>
  <arg name="urdf_file" default="$(find xacro)/xacro --inorder '$(find wheelchair_description)/urdf/wheelchair.xacro'" />
  <arg name="multi_robot_name" default=""/>
  <param name="robot_description" command="$(arg urdf_file)" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="22.0" />
    <param name="tf_prefix" value="$(arg multi_robot_name)"/>
  </node>

  <!-- cartographer_node -->
  <node pkg="cartographer_ros" type="cartographer_node" name="cartographer_node" args="-configuration_directory $(find wheelchair_cartographer)/config -configuration_basename $(arg configuration_basename)" output="screen">
    <!-- <remap from="/imu" to="/flat_imu"/> -->
    <!-- <remap from="/scan" to="scan"/> -->
    <!--remap from="points2" to="/camera/depth/points" / -->
  </node>

  <!-- cartographer_occupancy_grid_node -->
  <node pkg="cartographer_ros" type="cartographer_occupancy_grid_node" name="cartographer_occupancy_grid_node" args="-resolution 0.05" />

  <!-- flat_world_imu_node-->
  <!-- <node pkg="wheelchair_cartographer" type="flat_world_imu_node" name="flat_world_imu_node" output="screen">
    <remap from="imu_in" to="/imu" />
    <remap from="imu_out" to="/flat_imu" />
  </node> -->

  <!-- rviz-->
  <group if="$(arg open_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find wheelchair_cartographer)/rviz/cartographer.rviz"/>
  </group>
</launch>
